rbx_types/src/tags.rs

```diff
@@ -0,0 +1,108 @@
+use std::convert::TryFrom;
+use std::str::FromStr;
+use std::string::FromUtf8Error;
+
+#[derive(Default, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(transparent)
+)]
+pub struct Tags {
+    tag_list: Vec<String>,
+}
+
+impl Tags {
+    pub fn new() -> Tags {
+        Self {
+            tag_list: Vec::new(),
+        }
+    }
+
+    pub fn as_vec(&self) -> &Vec<String> {
+        &self.tag_list
+    }
+
+    pub fn as_mut_vec(&mut self) -> &mut Vec<String> {
+        &mut self.tag_list
+    }
+}
+
+impl From<Vec<String>> for Tags {
+    fn from(tag_list: Vec<String>) -> Tags {
+        Self { tag_list }
+    }
+}
+
+impl From<Vec<&str>> for Tags {
+    fn from(tag_list: Vec<&str>) -> Tags {
+        Self {
+            tag_list: tag_list
+                .iter()
+                .map(|tag_name| String::from_str(tag_name).unwrap())
+                .collect(),
+        }
+    }
+}
+
+impl From<&Tags> for Vec<u8> {
+    fn from(tags: &Tags) -> Vec<u8> {
+        let tag_list = &tags.tag_list;
+        let mut buf: Vec<u8> = Vec::with_capacity(tag_list.capacity() + tag_list.len());
+
+        for tag_name in tag_list.as_slice() {
+            buf.extend_from_slice(tag_name.as_bytes());
+            buf.extend_from_slice(b"\0")
+        }
+
+        buf
+    }
+}
```

Comment:
This feels pretty weird to have as part of `From`. If we want this method, we should write an explicit method for it.

This can also be a one-liner, which makes me think we should consider writing that inline when we need it in rbx_xml and rbx_binary:

```rust
fn encode_tags(tags: &[&str]) -> Vec<u8> {
    tags.join("\u{0000}").into_bytes()
}
```